hyperspy/drawing/_widgets/polygon.py

```diff
@@ -0,0 +1,304 @@
+            for polygon in self._vertices_lists:
+                self._plot_polygon(polygon)
+
+        handle_props = dict(color="blue")
+        props = dict(color="blue")
+
+        self._widget = PolygonSelector(
+            ax,
+            onselect=self._onselect,
+            useblit=self.blit,
+            handle_props=handle_props,
+            props=props,
+        )
+        self._finished = False
+
+        self.ax.figure.canvas.draw_idle()
+
+    def set_polygons(self, polygons):
+        """Function for deleting currently saved polygons and setting new ones."""
+
+        if self._vertices_plots:
+            for plot in self._vertices_plots:
+                plot.remove()
+        self._vertices_plots = []
+
+        self._vertices_lists = polygons.copy()
+
+        if self.ax is not None and self.is_on:
+            for polygon in polygons:
+                self._plot_polygon(polygon)
+            self.ax.figure.canvas.draw_idle()
+
+    def connect(self, ax):
+        super().connect(ax)
+        self._set_mpl_callbacks()
+
+    def _set_mpl_callbacks(self):
+        self.cids.append(
+            self.ax.figure.canvas.mpl_connect("button_press_event", self._click_event)
+        )
+        self.cids.append(
+            self.ax.figure.canvas.mpl_connect("key_press_event", self._key_event)
+        )
+
+    def get_polygons(self):
+        """Returns a list of lists, where each inner list contains `(x, y)` tuples
+        of the vertices of a single polygon. Only completed polygons are included.
+        The polygons are not closed."""
+        if self._finished:
+            return self._vertices_lists + [self._widget.verts.copy()]
+        else:
+            return self._vertices_lists
+
+    def _plot_polygon(self, polygon, selected_plot=False):
+        closed_polygon = zip(*(polygon + [polygon[0]]))
+        if not selected_plot:
+            plot = self.ax.plot(*closed_polygon, animated=self.blit)
+            self._vertices_plots.append(plot)
+        else:
+            self._selected_plot = self.ax.plot(
+                *closed_polygon, animated=self.blit, linewidth=3, color="red"
+            )
+
+    def _onselect(self, vertices):
+        if len(self._widget.verts) > 0:
+            self._finished = True
+            self.events.changed.trigger(self)
+
+            all_polygons = self.get_polygons()
+            all_x_coordinates = [x for polygon in all_polygons for x, y in polygon]
+            all_y_coordinates = [y for polygon in all_polygons for x, y in polygon]
+
+            bounding_box = (
+                min(all_x_coordinates),
+                max(all_x_coordinates),
+                min(all_y_coordinates),
+                max(all_y_coordinates),
+            )
+            tup = (
+                (bounding_box[0] + bounding_box[1]) / 2,
+                (bounding_box[2] + bounding_box[3]) / 2,
+            )
+
+            self.position = tup
+
+    def _click_event(self, event):
```

Comment:
This should be all handled in matplotlib? If some changes are required in matplotlib, then we should make a PR to matplotlib and make a copy of the updated `matplotlib.widgets.PolygonSelector` in hyperspy (see https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/hyperspy/external/matplotlib/widgets.py).